deploy/gateway/tests/snapshot_test.yaml

```diff
@@ -5,5 +5,9 @@ excludeTemplates:
   - tls-secret.yaml
 tests:
   - it: should render
+    set:
+      metrics:
+        podMonitor:
+          enabled: true
```

Comment:
I think it's better to create a different test here to make sure we don't create `PodMonitor` by default.